primitives/account/src/lib.rs

```diff
@@ -28,14 +34,13 @@ use sp_runtime_interface::pass_by::PassByInner;
 /// A fully Ethereum-compatible `AccountId`.
 /// Conforms to H160 address and ECDSA key standards.
 /// Alternative to H256->H160 mapping.
-#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Default)]
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Default, Hash)]
 #[derive(Encode, Decode, MaxEncodedLen, TypeInfo)]
 pub struct AccountId20(pub [u8; 20]);
 
 #[cfg(feature = "serde")]
 impl_serde::impl_fixed_hash_serde!(AccountId20, 20);
 
-#[cfg(feature = "std")]
```

Comment:
This is not a correct change. `H160::from_str` implementation is only available in `primitive-types` (that `sp-core` re-exports) if `fixed-hash/rustc-hex` feature is enabled, which is in turn enabled by `primitive-types/rustc-hex`, which `sp-core` only enables when `std` feature is used.

As the result this can (and does) fail for downstream users unless feature unification accidentally enables `fixed-hash/rustc-hex`. This should either be reverted or `sp-core` needs to enable `primitive-types/rustc-hex` unconditionally.